express_relay/sdk/python/express_relay_utils/express_relay_types.py

```diff
@@ -0,0 +1,400 @@
+from enum import Enum
+from pydantic import BaseModel, field_validator, model_validator
+from uuid import UUID
+import web3
+from typing import Union, Dict, Any
+from pydantic import Field
+from typing_extensions import Literal
+
+from eth_account.datastructures import SignedMessage
+
+
+def assert_hex_string(s: str):
+    """
+    Validates that a string is a valid hex string.
+
+    Args:
+        s: The string to validate as a hex string. Can be '0x'-prefixed.
+    """
+    if s.startswith("0x"):
+        s = s[2:]
+
+    s_set = set(s)
+    hex_set = set("0123456789abcdefABCDEF")
+
+    assert s_set.issubset(hex_set), "v must be a valid hex string"
+
+
+class HexString(BaseModel):
+    """
+    Args:
+        string: The string to validate as a hex string.
+    """
+
+    string: str
+
+    @field_validator("string")
+    @classmethod
+    def check_hex(cls, v: str):
+        assert_hex_string(v)
+
+        return v
+
+
+class Address(BaseModel):
+    """
+    Args:
+        address: The string to validate as an Ethereum address.
+    """
+
+    address: str
+
+    @field_validator("address")
+    @classmethod
+    def check_address(cls, v: str):
+        assert web3.Web3.is_address(v), "s must be a valid Ethereum address"
+        return v
+
+
+class Bytes32(BaseModel):
+    """
+    Args:
+        string: The hex string to validate as a 32-byte hash.
+    """
+
+    string: str
+
+    @field_validator("string")
+    @classmethod
+    def check_bytes32(cls, v: str):
+        assert_hex_string(v)
+        assert len(v.replace("0x", "")) == 64, "permission key must be 32 bytes long"
+        return v
+
+
+class Bid(BaseModel):
+    """
+    Args:
+        amount: The amount of the bid in wei.
+        calldata: The calldata for the contract call.
+        chain_id: The chain ID to bid on.
+        contract: The contract address to call.
+        permission_key: The permission key to bid on.
+    """
+
+    amount: int
+    calldata: HexString
+    chain_id: str
+    contract: Address
+    permission_key: HexString
+
+    def to_dict(self):
```

Comment:
There should be a better way to do this only by customizing the `amount` field serialization instead of specifying everything.

In general defining `to_dict` and `from_dict` seems wrong as we are defining additional methods for serialization instead of overriding the default behavior that pydantic provides using model_dump/model_validate.